File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
        Map <String, Integer> map = new HashMap<>();
        
        for(String key : keymap){
            for(int i = 0; i < key.length(); i++){
                String tmp = String.valueOf(key.charAt(i));
                int val = i + 1;
                map.put(tmp, Math.min(map.getOrDefault(tmp, Integer.MAX_VALUE), val));
            }
        }
                        
        for(int idx = 0; idx < targets.length; idx++){
            String target = targets[idx];
            int result = 0;
            
            for(String s : target.split("")){
                int val = map.getOrDefault(s, -1);
                if(val == -1){
                    result = val;
                    break;
                }
                
                result += val;
            }
            answer[idx] = result;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    
    tmp = "ABCDEFGHIJKLMNOPQRSTUVWYXZ"
    alpha = [i for i in tmp]
    val = [1000 for i in range(len(tmp))]
    dictKeys = dict(zip(alpha, val))
    
    for key in keymap:
        listS = list(key)
        for idx in range(len(listS)):
            idxVal = idx + 1
            s = listS[idx]
            dictKeys[s] = min(dictKeys[s], idxVal)
            
    for target in targets:
        result = 0
        for s in target:
            if dictKeys[s] == 1000:
                result = -1
                break
            result += dictKeys[s]
        answer.append(result)
        
    return answer